api/services/CronService.js

```diff
@@ -44,6 +45,92 @@ module.exports = {
+        promises.push(Dependency.options.classMethods.findByDependantForIndependentDownloads(hit.fields.package[0]));
+        promises[promises.length-1].then(function(rootPackages){
+                          rootPackageNames = _.map(rootPackages,function(package){
+                              return package.package_name;
+                          });
+                          indirect = false;
+                          j=i+1;
+                          while (!indirect && j<hits.length && hits[j].fields.ip_id[0] == hit.fields.ip_id[0]
+                            && new Date(hits[j].fields.datetime[0]).getTime()< (new Date(hit.fields.datetime[0]).getTime()+60000)){
+                            if(_.includes(rootPackageNames,hits[j].fields.package[0]))
+                            {
+                              indirectDownloads[hit.fields.package[0]] = indirectDownloads[hit.fields.package[0]]+1 || 1;
+                              indirect=true;
+                            }
+                            j+=1;
+                          }
+                          j=i-1;
+                          while (j>=0 && hits[j].fields.ip_id[0] == hit.fields.ip_id[0]
+                            && new Date(hits[j].fields.datetime[0]).getTime()+60000> (new Date(hit.fields.datetime[0]).getTime())
+                            && !(indirect)){
+                            if(_.includes(rootPackageNames,hits[j].fields.package[0]))
+                            {
+                              indirectDownloads[hit.fields.package[0]] = indirectDownloads[hit.fields.package[0]]+1 || 1;
+                              indirect=true;
+                            }
+                            j-=1;
+                          }
+                          if(!indirect){
+                            directDownloads[hit.fields.package[0]] = directDownloads[hit.fields.package[0]]+1 || 1;
+                          }
+                        })                    
+          });
+        //when all promises are resolved, proceed and scroll search results
+        Promise.all(promises).then(function(){
+          dateBadFormat = new Date(date=hits[1].fields.datetime[0])
+          date = dateFormat(dateBadFormat,"yyyy-mm-dd").toString();
+          return ElasticSearchService.scrollLastMonthDownloadsBulk(response,date,directDownloads,indirectDownloads,total,callback);
+        });
+
+  },
+  //write all splitted download counts to the database
+  writeSplittedDownloadCounts: function(date,directDownloads,indirectDownloads){
+      console.log("writing data");
+      return Package.findAll({
+        attributes: ['name']
+      }).then(function(packages) {
+        var records = _.map(packages, function(package) {
+          var totalDownloads = directDownloads[package.name]+indirectDownloads[package.name] || directDownloads[package.name] || indirectDownloads[package.name] || 0;
```

Comment:
- [ ] The`totalDownloads` variable seems unused, or am I missing something ?
Just delete it if it's unused